libr/util/logging.c

```diff
@@ -0,0 +1,157 @@
+/* radare - LGPL - Copyright 2007-2013 - pancake */
+
+#define LOGGING_CONFIGSTR_SIZE 1024
+#define LOGGING_OUTPUTBUF_SIZE 8192
+
+#include <r_core.h>
+#include <stdarg.h>
+
+static RLoggingFuncdef cb_logfunc_print = NULL; // Function to call when outputting log string
```

Comment:
globals are not really thread-safe.

Reply:
My C knowledge is pretty limited when it comes to async/threaded. Is there an existing example in R2 I can look at to see how its done? Quick google points me to mutexing or thread-local storage. I'm not seeing any TLS stuff in libr at a glance though...

Reply:
Moved the buffer into _r_logging_internal. I'll ask around if its worth using thread local `__thread`/`__declspec(thread)` for the other things. We'd need to made defines for it and make sure its available on all supported platforms.